PR #2804 broke builds with mpi-serial

#2804 added, among other things, a reference to `MPI_ADDRESS_KIND`. Builds with mpi-serial are now failing (example: `SMS_Mmpi-serial_P1x1.f09_g16.S.cheyenne_gnu`) with:

```
/gpfs/fs1/work/sacks/ctsm_code/current_branch1/cime/src/drivers/mct/main/component_mod.F90:66:32:

   integer (kind=MPI_ADDRESS_KIND) :: max_mpi_tag
                                1
Error: Symbol ‘mpi_address_kind’ at (1) has no IMPLICIT type
```

I see MPI_ADDRESS_KIND defined in `src/externals/mct/mpi-serial/mpif.F90`, but I'm thinking that the issue is that component_mod has:

```
#include <mpif.h>
```

but NOT

```
use mpi
```

Should this be changed to `use mpi`, or should mpi-serial be changed to move MPI_ADDRESS_KIND to the header file?

I'm labeling this as critical, because it's blocking our upgrade to a more recent cime version in CTSM.

@jhkennedy @jedwards4b @ekluzek @fischer-ncar

@jedwards4b I just tried that; it didn't work: During the link, it generates:

```
component_mod.o: In function `__component_mod_MOD_component_init_pre':
component_mod.F90:(.text+0x3279): undefined reference to `mpi_attr_get_'
```

--- a/scripts/lib/get_tests.py
+++ b/scripts/lib/get_tests.py
@@ -64,7 +64,6 @@ _CIME_TESTS = {
                              "IRT_N2.f19_g16_rx1.A",
                              "ERR.f45_g37_rx1.A",
                              "ERP.f45_g37_rx1.A",
-                             "SMS_D_Ln9.f19_g16_rx1.A",
                              "SMS_D_Ln9_Mmpi-serial.f19_g16_rx1.A",
                              "DAE.ww3a.ADWAV",
                              "PET_P4.f19_f19.A",

--- a/scripts/lib/get_tests.py
+++ b/scripts/lib/get_tests.py
@@ -65,6 +65,7 @@ _CIME_TESTS = {
                              "ERR.f45_g37_rx1.A",
                              "ERP.f45_g37_rx1.A",
                              "SMS_D_Ln9.f19_g16_rx1.A",
+                             "SMS_D_Ln9_Mmpi-serial.f19_g16_rx1.A",
                              "DAE.ww3a.ADWAV",
                              "PET_P4.f19_f19.A",
                              "PEM_P4.f19_f19.A",

--- a/src/drivers/mct/main/component_mod.F90
+++ b/src/drivers/mct/main/component_mod.F90
@@ -63,8 +63,6 @@ module component_mod
   integer  :: mpicom_GLOID, mpicom_CPLID           ! GLOID, CPLID mpi communicator
   integer  :: nthreads_GLOID, nthreads_CPLID
   logical  :: drv_threading
-  integer (kind=MPI_ADDRESS_KIND) :: max_mpi_tag
-
 
   !===============================================================================
 
@@ -105,9 +103,6 @@ contains
     call seq_comm_getinfo(GLOID, mpicom=mpicom_GLOID, iamroot=iamroot_GLOID, nthreads=nthreads_GLOID)
     call seq_comm_getinfo(CPLID, mpicom=mpicom_CPLID, iamroot=iamroot_CPLID, nthreads=nthreads_CPLID)
     iamin_CPLID = seq_comm_iamin(CPLID)
-
-    ! MPI spec. guarantees at least 32768 tags available, but most implementations will have more.
-    call mpi_attr_get(MPI_COMM_WORLD, MPI_TAG_UB, max_mpi_tag, flag, ierr)
 
     ! Initialize component type variables
     do eci = 1,size(comp)
@@ -332,7 +327,6 @@ contains
     type(mct_gGrid) :: dom_tmp   ! temporary
     character(*), parameter :: subname = '(component_init_cx)'
     character(*), parameter :: F0I = "('"//subname//" : ', A, 2i8 )"
-    character(*), parameter :: F1I = "('"//subname//" : ', A, I10, A, I10 )"
     !---------------------------------------------------------------
 
     ! Initialize driver rearrangers and AVs on driver
@@ -393,10 +387,6 @@ contains
                     mpi_tag = comp(eci)%cplcompid*100+eci*10+1
                 else
                     mpi_tag = comp(eci)%cplcompid*10000+eci*10+1
-                end if
-                if (mpi_tag > max_mpi_tag) then
-                   write(logunit, F1I) 'MPI message tag ',mpi_tag,' exceeds max ',max_mpi_tag
-                   call shr_sys_abort(subname//' ERROR: Max mpi tag exceeded')
                 end if
                 call seq_map_map_exchange(comp(1), flow='c2x', dom_flag=.true., msgtag=mpi_tag)
 
@@ -574,7 +564,6 @@ contains
     integer :: eci, num_inst
     integer :: mpi_tag
     character(*), parameter :: subname = '(component_init_areacor)'
-    character(*), parameter :: F1I = "('"//subname//" : ', A, I10, A, I10 )"
     !---------------------------------------------------------------
 
     num_inst = size(comp)
@@ -588,10 +577,6 @@ contains
              mpi_tag = comp(eci)%cplcompid*100+eci*10+5
           else
              mpi_tag = comp(eci)%cplcompid*10000+eci*10+5
-          end if
-          if (mpi_tag > max_mpi_tag) then
-             write(logunit, F1I) 'MPI message tag ',mpi_tag,' exceeds max ',max_mpi_tag
-             call shr_sys_abort(subname//' ERROR: Max mpi tag exceeded')
           end if
           call seq_map_map(comp(eci)%mapper_Cx2c, comp(eci)%dom_cx%data, comp(eci)%dom_cc%data, msgtag=mpi_tag)
 
@@ -615,10 +600,6 @@ contains
               mpi_tag = comp(eci)%cplcompid*100+eci*10+7
           else
               mpi_tag = comp(eci)%cplcompid*10000+eci*10+7
-          end if
-          if (mpi_tag > max_mpi_tag) then
-             write(logunit, F1I) 'MPI message tag ',mpi_tag,' exceeds max ',max_mpi_tag
-             call shr_sys_abort(subname//' ERROR: Max mpi tag exceeded')
           end if
           call seq_map_map(comp(eci)%mapper_cc2x, comp(eci)%c2x_cc, comp(eci)%c2x_cx, msgtag=mpi_tag)
 
@@ -856,7 +837,6 @@ contains
     integer :: ierr
     integer :: mpi_tag
     character(*), parameter :: subname = '(component_exch)'
-    character(*), parameter :: F1I = "('"//subname//" : ', A, I10, A, I10 )"
     !---------------------------------------------------------------
 
     if (present(timer_barrier))  then
@@ -885,20 +865,12 @@ contains
              else
                 mpi_tag = comp(eci)%cplcompid*10000+eci*10+2
              end if
-             if (mpi_tag > max_mpi_tag) then
-                write(logunit, F1I) 'MPI message tag ',mpi_tag,' exceeds max ',max_mpi_tag
-                call shr_sys_abort(subname//' ERROR: Max mpi tag exceeded')
-             end if
              call seq_map_map(comp(eci)%mapper_Cx2c, comp(eci)%x2c_cx, comp(eci)%x2c_cc, msgtag=mpi_tag)
           else if (flow == 'c2x') then ! component to coupler
              if ( size(comp) > 1) then
                 mpi_tag = comp(eci)%cplcompid*100+eci*10+4
              else
                 mpi_tag = comp(eci)%cplcompid*10000+eci*10+4
-             end if
-             if (mpi_tag > max_mpi_tag) then
-                write(logunit, F1I) 'MPI message tag ',mpi_tag,' exceeds max ',max_mpi_tag
-                call shr_sys_abort(subname//' ERROR: Max mpi tag exceeded')
              end if
              call seq_map_map(comp(eci)%mapper_Cc2x, comp(eci)%c2x_cc, comp(eci)%c2x_cx, msgtag=mpi_tag)
           end if

